plugins/database/postgresql/postgresql.go

```diff
@@ -104,54 +107,56 @@ func (p *PostgreSQL) getConnection(ctx context.Context) (*sql.DB, error) {
 	return db.(*sql.DB), nil
 }
 
-// SetCredentials uses provided information to set/create a user in the
-// database. Unlike CreateUser, this method requires a username be provided and
-// uses the name given, instead of generating a name. This is used for creating
-// and setting the password of static accounts, as well as rolling back
-// passwords in the database in the event an updated database fails to save in
-// Vault's storage.
-func (p *PostgreSQL) SetCredentials(ctx context.Context, statements dbplugin.Statements, staticUser dbplugin.StaticUserConfig) (username, password string, err error) {
-	if len(statements.Rotation) == 0 {
-		statements.Rotation = []string{defaultPostgresRotateRootCredentialsSQL}
+func (p *PostgreSQL) UpdateUser(ctx context.Context, req newdbplugin.UpdateUserRequest) (newdbplugin.UpdateUserResponse, error) {
+	if req.Username == "" {
+		return newdbplugin.UpdateUserResponse{}, fmt.Errorf("missing username")
+	}
+	if req.Password == nil && req.Expiration == nil {
+		return newdbplugin.UpdateUserResponse{}, fmt.Errorf("no changes requested")
 	}
 
-	username = staticUser.Username
-	password = staticUser.Password
-	if username == "" || password == "" {
-		return "", "", errors.New("must provide both username and password")
+	merr := &multierror.Error{}
+	if req.Password != nil {
+		merr = multierror.Append(merr, p.changeUserPassword(ctx, req.Username, req.Password))
```

Comment:
Nit: This is a bit more verbose, but having the call to `p.changeUserPassword` outside of `multierror.Append` makes it a bit easier read.

```suggestion
		err := p.changeUserPassword(ctx, req.Username, req.Password)
		merr = multierror.Append(merr, err)
```

Reply:
Good call. Done